src/prefect/cli/server.py

```diff
@@ -294,6 +279,41 @@ def start(
                 " run `prefect server stop`."
             )
 
+    # Check if host/port were explicitly passed (not using default values)
+    current_profile = load_current_profile()
```

Comment:
Why are you using `load_current_profile` instead of `get_current_settings`?

Reply:
I probably shouldn't be, but I did because of the tests that were checking for specific profile switching logic

... I'm not sure that this PR needs to exist after https://github.com/PrefectHQ/prefect/issues/17456#issuecomment-2725666438